### PR TITLE
chore(elixir): add logging telemetry events

### PR DESCRIPTION
**Pull request by wojciech-wais**
## Current Behaviour
@hairyhum 

Some time ago you have suggested the solution for retrieving all modules that implement `Ockam.Worker`.

I got back to the solution that you have suggested.
## Proposed Changes

I am continuing the work which was started in #2212 

## Checks

- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits).
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://www.ockam.io/learn/how-to-guides/contributing/CONTRIBUTING#commit-messages).
- [x] I accept the Ockam Community [Code of Conduct](https://www.ockam.io/learn/how-to-guides/high-performance-team/conduct/).
- [x] I have accepted the Ockam [Contributor License Agreement](https://www.ockam.io/learn/how-to-guides/contributing/cla/) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/contributors](https://github.com/build-trust/contributors) repository.

### CODE REVIEW

**Comment by hairyhum** [comment]
We have `[Ockam.Worker, :init, :start]` and `[Ockam.Worker, :init, :stop]` events now which report module name in the metadata (also for `:handle_message`). I don't think using module-specific events is a good idea, since it's not easy to get them.

I don't think we should use them, and we might even delete them in the future.

https://github.com/build-trust/ockam/blob/develop/implementations/elixir/ockam/ockam/lib/ockam/worker.ex#L236

**Comment by hairyhum** [comment]
Also interesting addition is `Ockam.Node.list_workers()` which gets all currently running workers with their address, pid and module.

**Comment by wojciech-wais** [comment]
> We have `[Ockam.Worker, :init, :start]` and `[Ockam.Worker, :init, :stop]` events now which report module name in the metadata (also for `:handle_message`). I don't think using module-specific events is a good idea, since it's not easy to get them.
> 
> I don't think we should use them, and we might even delete them in the future.
> 
> https://github.com/build-trust/ockam/blob/develop/implementations/elixir/ockam/ockam/lib/ockam/worker.ex#L236

@hairyhum 

Ok. So now I set the handler for those two types of events:
`[Ockam.Worker, :init, :start]` and `[Ockam.Worker, :init, :stop]`

I have also added output from `Ockam.Node.list_workers()` into the logger message, but it floods the log output like that and I am wondering if this is the thing that you wanted to achieve.

Maybe I should just print worker address, pid and nothing more?
How do you think? 

`===> 
[:ockam, Ockam.Worker, :init, :stop], 
%{duration: 974743}, 
%{address: "printer", module: Ockam.Router.Tests.Printer, options: [address: "printer"], result: :ok, return_value: {:ok, %{address: "printer", last_message_ts: nil, module: Ockam.Router.Tests.Printer}}}, 
[{"08d55a147816154d", #PID<0.1016.0>, Ockam.Stream.Index.Shard}, {"2c67cbbe45fe916a", #PID<0.1076.0>, Ockam.SecureChannel.Channel}, {"ST_S_b3c08c44f4327400", #PID<0.991.0>, Ockam.Stream.Workers.Stream}, {"0e3eb2f34d262d47", #PID<0.1075.0>, nil}, {"S_I_3fd57426b574d0fc", #PID<0.1021.0>, Ockam.Session.Pluggable.Initiator}, ..........................
 {"ST_S_3af21e413263c36f", #PID<0.998.0>, Ockam.Stream.Workers.Stream}, {"c180cf00da514256", #PID<0.1010.0>, Ockam.Stream.Index.Service}, {"17c6145f80eddbd7", #PID<0.1089.0>, nil}, {"S_R_12af99ff8272d39d", #PID<0.1042.0>, Ockam.Session.Pluggable.Responder}, {"explicit_inner", #PID<0.1125.0>, Ockam.Worker.Tests.InnerWorker}, {"774767d75972d307", #PID<0.1037.0>, Ockam.Messaging.Delivery.ResendPipe.Sender}, {"855c43f968094e9b", #PID<0.1080.0>, Ockam.SecureChannel.Channel}, {"2f43ff65b4e3677b", #PID<0.1088.0>, ...}, {"108dcb1561e7574d", ...}, {...}, ...]
`

File: implementations/elixir/ockam/ockam/lib/ockam.ex [hunk 1]
```diff
@@ -15,6 +15,7 @@ defmodule Ockam do
   def start(_type, _args) do
     if Code.ensure_loaded?(:telemetry) do
       {:ok, _apps} = Application.ensure_all_started(:telemetry)
+      Ockam.TelemetryLogger.attach()
```
Comment (hairyhum) [inline implementations/elixir/ockam/ockam/lib/ockam.ex]: Let's keep it detached by default.

File: implementations/elixir/ockam/ockam/lib/ockam/telemetry_logger.ex [hunk 2]
```diff
@@ -6,21 +6,29 @@ if Code.ensure_loaded?(:telemetry) do
     require Logger
 
     def handle_event(event, measurements, metadata, _config) do
+      ockam_workers = Ockam.Node.list_workers()
```
Comment (hairyhum) [inline implementations/elixir/ockam/ockam/lib/ockam/telemetry_logger.ex]: Let's not add this to the log, it's not that helpful.

**Review by hairyhum — CHANGES_REQUESTED** [review]

File: implementations/elixir/ockam/ockam/lib/ockam.ex (hunk 1 above)
Comment (wojciech-wais) [reply implementations/elixir/ockam/ockam/lib/ockam.ex]: Ok. Fixed. Not attached now.

File: implementations/elixir/ockam/ockam/lib/ockam/telemetry_logger.ex (hunk 2 above)
Comment (wojciech-wais) [reply implementations/elixir/ockam/ockam/lib/ockam/telemetry_logger.ex]: Ok. Removed from log message.

**Review by hairyhum — DISMISSED** [review]

**Comment by mrinalwadhwa** [comment]
@feniks65 thank you for another awesome contribution! 🎉 
I rebased and merged in https://github.com/build-trust/ockam/pull/2823

**Pull request closed without merging**